### PR TITLE
Say that shaders should terminate, or UA may lose the device

### PR DESCRIPTION
**Pull request by dneto0**
Fixes: #1987

### CODE REVIEW

**Review by alan-baker — COMMENTED** [review]
LGTM, but leaving approval for API editors.

**Review by toji — APPROVED** [review]
LGTM. Covers the Chrome GPU watchdog behavior, at least, and leaves room for other implementations to take a less disruptive path if they can.

**Comment by dneto0** [comment]
Hey @toji should this land or go on the tacit resolution queue?

**Comment by toji** [comment]
Probably the tacit resolution queue. While I think the behavior specified is straightforward enough and I have a hard time imagining anyone disagreeing with it, it's not a strictly editorial change so we should at least bring it to the group's attention.

File: spec/index.bs [hunk 1]
```diff
@@ -12176,6 +12176,9 @@ These operations are encoded within {{GPUComputePassEncoder}} as:
 
 Issue: describe the computing algorithm
 
+If [[WGSL#shader-execution-end|shader execution does not end]],
```
Comment (litherum) [inline spec/index.bs:12179]: Perhaps "in a reasonable amount of time"? (and leave "reasonable" up to the UA)

Comment (kainino0x) [reply spec/index.bs:12179]: I'd probably write:
```
If [[WGSL#shader-execution-end|shader execution end]] does not occur in a reasonable amount of time,
```
which uses the "shader execution end" wording from WGSL verbatim.

File: spec/index.bs [hunk 2]
```diff
@@ -12176,6 +12176,9 @@ These operations are encoded within {{GPUComputePassEncoder}} as:
...
+then a {{GPUError}} may be issued, or the [=device=] may become [=lose the device|lost=].
```
Comment (kainino0x) [inline spec/index.bs:12180]: If we allow a GPUError, we should be specific about which one (maybe #3123).

However if we allow that we also need to be very precise about what does and doesn't happen. Does the next call in the pass happen? Is the whole pass cancelled? Is the whole command buffer cancelled? In any of those cases, do the stores at the end of the render pass happen?

I suspect there are no strong guarantees from underlying APIs in cases like this and we should simply allow only device lost for now.

**Comment by kainino0x** [comment]
Moving out of tacit resolution queue because editors don't have an agreed-upon resolution yet.

File: spec/index.bs (hunk 2 above)
Comment (dneto0) [reply spec/index.bs:12180]: Right.  I was guessing.   From what I know, it's probably only context lost.  I'll remove the GPUError and see if that's more palatable.

**Pull request merged**